lib/sycamore/sycamore/tests/integration/scans/test_weaviate_scan.py

```diff
@@ -0,0 +1,98 @@
+    with weaviate.WeaviateClient(**weaviate_client_args) as client:
+        time.time()
+        timeout = 30
+        deadline = time.time() + timeout
+        while not client.is_live():
+            time.sleep(1)
+            if time.time() > deadline:
+                raise RuntimeError(f"Weaviate failed to start in {timeout} seconds")
+        yield {
+            "connection_params": ConnectionParams.from_params(
+                http_host="localhost",
+                http_port=port,
+                http_secure=False,
+                grpc_host="localhost",
+                grpc_port=grpc_port,
+                grpc_secure=False,
+            ),
+        }
+
+
+def test_weaviate_scan(wv_client_args):
+
+    paths = str(TEST_DIR / "resources/data/pdfs/Transformer.pdf")
+    model_name = "sentence-transformers/all-MiniLM-L6-v2"
+    tokenizer = HuggingFaceTokenizer(model_name)
+    collection = "TestCollection"
+    collection_config_params = {
+        "name": collection,
+        "description": "A collection to demo data-prep with sycamore",
+        "properties": [
+            Property(
+                name="properties",
+                data_type=DataType.OBJECT,
+                nested_properties=[
+                    Property(
+                        name="links",
+                        data_type=DataType.OBJECT_ARRAY,
+                        nested_properties=[
+                            Property(name="text", data_type=DataType.TEXT),
+                            Property(name="url", data_type=DataType.TEXT),
+                            Property(name="start_index", data_type=DataType.NUMBER),
+                        ],
+                    ),
+                ],
+            ),
+            Property(name="bbox", data_type=DataType.NUMBER_ARRAY),
+            Property(name="shingles", data_type=DataType.INT_ARRAY),
+        ],
+        "vectorizer_config": [Configure.NamedVectors.none(name="embedding")],
+        "references": [ReferenceProperty(name="parent", target_collection=collection)],
+    }
+
+    ctx = sycamore.init()
+
+    docs = (
+        ctx.read.binary(paths, binary_format="pdf")
+        .partition(partitioner=UnstructuredPdfPartitioner())
+        .regex_replace(COALESCE_WHITESPACE)
+        .mark_bbox_preset(tokenizer=tokenizer)
+        .merge(merger=MarkedMerger())
+        .spread_properties(["path"])
+        .split_elements(tokenizer=tokenizer, max_tokens=512)
+        .explode()
+        .embed(embedder=SentenceTransformerEmbedder(model_name=model_name, batch_size=100))
+        .sketch(window=17)
+        .take_all()
+    )
+    ctx.read.document(docs).write.weaviate(
+        wv_client_args=wv_client_args, collection_name=collection, collection_config=collection_config_params
+    )
+
+    out_docs = ctx.read.weaviate(wv_client_args=wv_client_args, collection_name=collection).take_all()
+    assert len(out_docs) == len(docs)
+    assert {d.doc_id for d in docs} == {d.doc_id for d in out_docs}
```

Comment:
Would suggest testing for full equality here, as done below: https://github.com/aryn-ai/sycamore/blob/d51348f6ebfcdbc487804057e1a572f208694b10/lib/sycamore/sycamore/tests/integration/scans/test_db_scan.py#L96